test/integration/runner-spec/iterationCount.test.js

```diff
@@ -0,0 +1,230 @@
+                    passed: false
+                });
+                expect(assertions[2]).to.deep.include({
+                    name: 'should contain data',
+                    passed: true
+                });
+                expect(assertions[3]).to.deep.include({
+                    name: 'should contain data',
+                    passed: true
+                });
+            });
+        });
+
+        describe('with data items more than iteration count', function () {
+            var testrun;
+
+            before(function (done) {
+                this.run({
+                    iterationCount: 2,
+                    data: [
+                        {foo: 'bar'},
+                        {foo: 'bar'},
+                        {foo: 'bar'}
+                    ],
+                    collection
+                }, function (err, results) {
+                    testrun = results;
+                    done(err);
+                });
+            });
+
+            it('should complete the run successfully', function () {
+                expect(testrun).to.be.ok;
+                expect(testrun.done.getCall(0).args[0]).to.not.exist;
+                expect(testrun).to.nested.include({
+                    'done.calledOnce': true,
+                    'start.calledOnce': true
+                });
+            });
+
+            it('should ignore the data items after the iteration count', function () {
+                var assertions = [
+                    testrun.assertion.getCall(0).args[1][0],
+                    testrun.assertion.getCall(1).args[1][0]
+                ];
+
+                expect(assertions[0]).to.deep.include({
+                    name: 'should contain data',
+                    passed: true
+                });
+                expect(assertions[1]).to.deep.include({
+                    name: 'should contain data',
+                    passed: true
+                });
+            });
+        });
+    });
+
+    describe('when not set', function () {
+        describe('with data', function () {
+            var testrun;
+
+            before(function (done) {
+                this.run({
+                    data: [
+                        {foo: 'bar'},
+                        {foo: 'bar'},
+                        {foo: 'bar'}
+                    ],
+                    collection
+                }, function (err, results) {
+                    testrun = results;
+                    done(err);
+                });
+            });
+
+            it('should complete the run successfully', function () {
+                expect(testrun).to.be.ok;
+                expect(testrun.done.getCall(0).args[0]).to.not.exist;
+                expect(testrun).to.nested.include({
+                    'done.calledOnce': true,
+                    'start.calledOnce': true
+                });
+            });
+
+            it('should set iterations equal to the length of data', function () {
```

Comment:
divide into:
- `it('should run iterations equal to the length of data')`
- `it('should use proper data in each iterations')`